### PR TITLE
 partially monomorphise & de-generify Env

### PR DESCRIPTION
**Pull request by NadiaYvette**
Store's User case, Action's Set case & branching through Name's cases within Store's Named case are all unfolded. The entire Setters file along with its Tag, Sum etc. data types are removed.

### CODE REVIEW

File: bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs [hunk 1]
```diff
@@ -42,10 +44,12 @@ import           Cardano.Benchmarking.Script.Store
...
 import           Cardano.TxGenerator.Types (TxGenError(..))
 
-data Env = Env { dmap :: DMap Store Identity }
+data Env = Env { dmap :: DMap Store Identity
+               , protoParams :: Maybe ProtocolParameterMode
+               }
 
 emptyEnv :: Env
-emptyEnv = Env { dmap = DMap.empty }
+emptyEnv = Env { dmap = DMap.empty, protoParams = Nothing }
```
Comment (deepfire) [inline bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs:52]: Just a very minor nitpick about formatting -- I'd:

0. break the line, between `=` and `Env` aligning the latter naturally.
1. break the line, just before the `,`  aligning the latter with the `{`
2. make the trailing `}` on its own line

...which is how multi-line is done frequently.

File: bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs [hunk 2]
```diff
@@ -73,12 +77,21 @@ askIOManager = lift RWS.ask
 set :: Store v -> v -> ActionM ()
 set key val = lift $ RWS.modify $ (\e -> e { dmap = DMap.insert key (pure val) (dmap e)})
 
+setProtoParamMode :: ProtocolParameterMode -> ActionM ()
+setProtoParamMode val = lift $ RWS.modify $ (\e -> e { protoParams = pure val })
```
Comment (deepfire) [inline bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs:81]: Given that you'll have more of those cases, I'd split this into a combination of:

1. the generic `lift . RWS.modify` helper and have it accept the: 
2. the `Env` modifier for specific cases

File: bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs [hunk 3]
```diff
@@ -73,12 +77,21 @@ askIOManager = lift RWS.ask
...
+
 get :: Store v -> ActionM v
 get key = do
   lift (RWS.gets $ (\e -> DMap.lookup key $ dmap e)) >>= \case
     Just (Identity v) -> return v
     Nothing -> throwE $ LookupError key
 
+getProtoParamMode :: ActionM ProtocolParameterMode
+getProtoParamMode = do
+  lift (RWS.gets $ (\e -> protoParams e)) >>= \case
```
Comment (deepfire) [inline bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs:91]: Here, we can similarly to the setter, extract a generic helper that calls a supplied reader on top of the common `lift . RWS.gets` part.

Comment (NadiaYvette) [reply bench/tx-generator/src/Cardano/Benchmarking/Script/Env.hs:100]: I'll mop up the read accessors in a sweep after finishing off all the Store cases. The set accessors might still be worthwhile vs. open-coding, though lift $ modify lambda isn't too big a deal either.

**Review by mgmeier — APPROVED** [review]

**Comment by mgmeier** [comment]
Thank you a lot, Nadia! :)

**Pull request merged**